core/src/main/java/com/valtech/aemsaas/core/services/configuration/BaseSaasConfiguration.java

```diff
@@ -0,0 +1,15 @@
+package com.valtech.aemsaas.core.services.configuration;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.caconfig.annotation.Configuration;
+import org.apache.sling.caconfig.annotation.Property;
+
+@Configuration(label = "Base AEM Saas - Configuration", description = "Base AEM Saas Configuration")
```

Comment:
let's add some more description, e.g. `Base Search Configuration is used for all search requests done for the defined content tree.`